### ISSUE TITLE
Site editor: Site icon moves off-screen when tabbing through the UI

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#51257]
### Description

Turns out the addition of the 'View site' link to the site hub in the site editor, see https://github.com/WordPress/gutenberg/pull/50420, broke the layout when tabbing with teh keyboard trough the UI. See animated GIF:

![tabbing](https://github.com/WordPress/gutenberg/assets/1682452/1c8f407a-1efd-4eff-adce-522776e321e8)

When the navigation panel is closed, the 'View site' link is stil focusable and can be tabbed to. Instead, I think it should not be rendered. It should work like the 'Open command center' button, which is rendered conditionally only in the 'view' mode. See https://github.com/WordPress/gutenberg/blob/b30e1efb3415a8d43a82a9c735078fdbd351105e/packages/edit-site/src/components/site-hub/index.js#L159-L175
Cc @glendaviesnz 

### Step-by-step reproduction instructions

- Go to the Site editor.
- Use the Tab key to navigate to the canvas overlay.
- Press Enter to switch the editor to Edit mode.
- Press the Tab key once: focus goes to the first block. Not sure this is OK but it's a separate issue.
- Keep pressing the Tab key to navigate through the entire UI and start from scratch.
- Or Press Shift + Tab to navigate backwards.
- Observe that the 'View site' link is still focusable. It stays not visible, only the tooltip is visible.
- Observe that when the 'View site' link receives focus, the Site icon button moves off-screen and the layout breaks.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by glendaviesnz** [comment]
Thanks @afercia for spotting this and suggesting a fix. I have added a [PR here](https://github.com/WordPress/gutenberg/pull/51279).
